File: samples/donkey_kong/data/scripts/dk/height.py
import dk.level1
import dk.sprites
import proto.animation_pb2
import pytroll.audio
import pytroll.scene


class HeightScene(pytroll.scene.PyTrollScene):
    def __init__(self):
        self.scene = proto.scene_pb2.Scene()
        self.scene.id = 'height'
        self.scene.viewport.width = 640
        self.scene.viewport.height = 480

    def Setup(self):
        donkey_kong = dk.sprites.DonkeyKong('dk')
        donkey_kong.Create((300, 382, 0), frame_index=10)
        pytroll.audio.PlayMusic('height')

        script = proto.animation_pb2.AnimationScript()
        script.id = 'height'
        script.animation.add().timer.delay = 3500
        donkey_kong.PlayAnimationScript(
            script, lambda: self.Transition(dk.level1.Level1Scene()))

File: samples/dart_dk/lib/height.dart
import 'package:dart_dk/level1.dart';
import 'package:dart_dk/src/sprites/donkey_kong.dart';
import 'package:dart_troll/src/core/audio.dart';
import 'package:dart_troll/src/core/scene.dart';
import 'package:dart_troll/src/proto/animation.pb.dart';
import 'package:dart_troll/src/proto/primitives.pb.dart';
import 'package:dart_troll/src/proto/scene.pb.dart' as proto;

class HeightScene extends Scene {
  @override
  proto.Scene sceneDefinition() {
    return proto.Scene()
      ..id = 'height'
      ..viewport = (Box()
        ..width = 640
        ..height = 480);
  }

  @override
  void setup() {
    DonkeyKong([300, 282], frameIndex: 10)
      ..playAnimation(
          script: (AnimationScript()
            ..id = 'height'
            ..animation.addAll([
              Animation()..timer = (TimerAnimation()..delay = 3500),
            ])),
          onDone: (Event) => transition(Level1Scene()));

    playMusic('height');
  }
}
